vterm.el

```diff
@@ -529,29 +549,33 @@ Feeds the size change to the virtual terminal."
   (let ((succ (eq 0 (forward-line (1- n)))))
     succ))
 
-(defun vterm--buffer-line-num()
-  "Return the maximum line number."
-  (count-lines (point-min) (point-max)))
-
 (defun vterm--set-title (title)
   "Run the `vterm--set-title-hook' with TITLE as argument."
   (run-hook-with-args 'vterm-set-title-functions title))
 
 (defun vterm--set-directory (path)
   "Set `default-directory' to PATH."
-  (if (string-match "^\\(.*?\\)@\\(.*?\\):\\(.*?\\)$" path)
-      (progn
-        (let ((user (match-string 1 path))
-              (host (match-string 2 path))
-              (dir (match-string 3 path)))
-          (if (and (string-equal user user-login-name)
-                   (string-equal host (system-name)))
-              (progn
-                (when (file-directory-p dir)
-                  (setq default-directory dir)))
-            (setq default-directory (concat "/-:" path)))))
-    (when (file-directory-p path)
-      (setq default-directory path))))
+  (let ((dir (vterm--get-directory path)))
+    (when dir (setq default-directory dir))))
+
+(defun vterm--get-directory (path)
+  "Get  normalized directory to PATH."
+  (when path
```

Comment:
This should not be needed, does it?

Reply:
this is used in `vterm-next-error-function`